cloudflare_utils.py

```diff
@@ -9,12 +9,36 @@
 
 def get_cloudflare_ip_ranges():
     cloudflare_ip_ranges_url = 'https://www.cloudflare.com/ips-v4'
+    ip_ranges = []
+    ip_ranges_fallback = [
+        "103.21.244.0/22",
+        "103.22.200.0/22",
+        "103.31.4.0/22",
+        "104.16.0.0/12",
+        "108.162.192.0/18",
+        "131.0.72.0/22",
+        "141.101.64.0/18",
+        "162.158.0.0/15",
+        "172.64.0.0/13",
+        "173.245.48.0/20",
+        "188.114.96.0/20",
+        "190.93.240.0/20",
+        "197.234.240.0/22",
+        "198.41.128.0/17"
+    ]
+
     try:
+        print('[*] Retrieving Cloudflare IP ranges from https://www.cloudflare.com/ips-v4')
         page_content = requests.get(cloudflare_ip_ranges_url, timeout=10)
         ip_ranges_text = page_content.text
-        return [ip for ip in ip_ranges_text.split("\n") if ip]
+        ip_ranges = [ip for ip in ip_ranges_text.split("\n") if ip]
+
     except requests.exceptions.RequestException:
-        pass
+        sys.stderr.write('Failed to retrieve Cloudflare IP ranges - using a default (possibly outdated) list')
```

Comment:
You're missing a `\n` at the end (which `sys.stderr.write` doesn't append)

Reply:
And maybe add a `[-]` before the error message, so we more easily see it's an error :)

Reply:
Ah, you're right. Fixed.